src/jsr.rs

```diff
@@ -349,14 +375,36 @@ mod test {
   #[test]
   fn test_jsr_dep_pkg_req_serializable() {
     fn run_test(text: &str) {
-      let start = JsrDepPackageReq::from_str(text).unwrap();
+      let start = JsrDepPackageReq::from_str_loose(text).unwrap();
       let value = serde_json::to_value(&start).unwrap();
       let deserialized: JsrDepPackageReq =
         serde_json::from_value(value).unwrap();
       assert_eq!(deserialized, start);
     }
 
     run_test("jsr:a@1");
-    run_test("npm:a@1")
+    run_test("npm:a@1");
+    run_test("npm:a@1 - 1.5");
+    run_test("npm:a@1 || 2 || 3");
```

Comment:
Just a thought, but would it make sense to enclose the version constraint in a single quote to make parsing/reading easier? So:
```
    run_test("npm:a@'1 - 1.5'");
    run_test("npm:a@'1 || 2 || 3'");
```
It might not matter since it's an internal format and these constraints can't be used in import specifiers either, right?